dateutil/test/test_isoparser.py

```diff
@@ -453,11 +461,18 @@ def __make_time_examples():
 @pytest.mark.parametrize('time_val,time_fmt', __make_time_examples())
 @pytest.mark.parametrize('as_bytes', [True, False])
 def test_isotime(time_val, time_fmt, as_bytes):
-    tstr = time_val.strftime(time_fmt)
-    if isinstance(time_val, six.text_type) and as_bytes:
-        tstr = tstr.encode('ascii')
-    elif isinstance(time_val, bytes) and not as_bytes:
-        tstr = tstr.decode('ascii')
+    if six.PY2:
```

Comment:
As with the other changes, this is not the correct approach.

Reply:
In this case, the problem is that in the original code, I was using:

```python
if isinstance(time_val, six.text_type) and as_bytes:
    ...
```
But what I *wanted* was:

```python
if isinstance(tstr, six.text_type) and as_bytes:
    ...
```
Because `time_val` is the time input, not the output of `time_val.strftime` (which will either be `bytes` or `str` depending on the Python version).